sandbox/web_server/src/bin/main.rs

```diff
@@ -1,24 +1,79 @@
 use std::io::prelude::*;
 use std::net::TcpStream;
 use std::net::TcpListener;
-use web_server::ThreadPool;
+use web_server::{ThreadPool, Cell};
+use std::sync::{Arc, Mutex, mpsc};
+use rand::prelude::*;
+
+const SIZE: usize = 64;
 
 fn main() {
     let listener = TcpListener::bind("127.0.0.1:7878").unwrap();
-    let pool = ThreadPool::new(4);
+    let buffer = Arc::new(Mutex::new([Cell { plain: 0, key: 0 }; SIZE]));
+
+    let counter = Arc::new(Mutex::new(1));
+    let pool = ThreadPool::new(SIZE);
+    let (sender, receiver) = mpsc::channel();
+    let receiver = Arc::new(Mutex::new(receiver));
+    let sender = Arc::new(Mutex::new(sender.clone()));
 
     for stream in listener.incoming() {
         let stream = stream.unwrap();
-        pool.execute(|| {
-            handle_connection(stream);
+        let sender = Arc::clone(&sender);
+        let receiver = Arc::clone(&receiver);
+        let counter = Arc::clone(&counter);
+        let buffer = Arc::clone(&buffer);
+        pool.execute(move || {
+            handle_connection(stream, counter, buffer, sender, receiver);
         });
     }
 }
 
 
-fn handle_connection(mut stream: TcpStream) {
+pub fn handle_connection(mut stream: TcpStream, counter: Arc<Mutex<i32>>, buf: Arc<Mutex<[Cell; SIZE]>>, sen: Arc<Mutex<mpsc::Sender<()>>>, rec: Arc<Mutex<mpsc::Receiver<()>>>) {
     let mut buffer = [0; 512];
     stream.read(&mut buffer).unwrap();
-    println!("Request: {}", String::from_utf8_lossy(&buffer[..]).to_ascii_lowercase());
+    //get the index of the first nullbyte
+    let mut j = 0;
+    for i in 0..512 {
+        if buffer[i] == 0 {
+            break;
+        } else { j += 1; }
+    }
+    let plain_string: String = String::from_utf8_lossy(&buffer[..j]).to_string();
 
-}
+    let plain: u64 = plain_string.parse().unwrap();
+    let key = rand::thread_rng().gen();
+
+    let mut cpt = counter.lock().unwrap();
+    let number = *cpt;
```

Comment:
Make sure that `number` is always in the interval [0, SIZE]. Remove occurrences of `number % ...`